deeprank2/query.py

```diff
@@ -716,40 +585,40 @@ def build(self, feature_modules: List[ModuleType], include_hydrogens: bool = Fal
             :class:`Graph`: The resulting :class:`Graph` object with all the features and targets.
         """
 
-        contact_atoms = _load_ppi_atoms(self._pdb_path,
-                                        self._chain_id1, self._chain_id2,
-                                        self._distance_cutoff,
+        contact_atoms = _load_ppi_atoms(self.pdb_path,
+                                        self.chain_ids[0], self.chain_ids[1],
+                                        self.distance_cutoff,
                                         include_hydrogens)
 
-        atom_positions = []
-        residues_selected = set([])
-        for atom in contact_atoms:
-            atom_positions.append(atom.position)
-            residues_selected.add(atom.residue)
-        residues_selected = list(residues_selected)
-
         # build the graph
-        graph = build_residue_graph(
-            residues_selected, self.get_query_id(), self._distance_cutoff
-        )
+        if self.resolution == 'atomic':
+            graph = build_atomic_graph(contact_atoms, self.get_query_id(), self.distance_cutoff)
+        elif self.resolution == 'residue':
+            residues_selected = {atom.residue for atom in contact_atoms}
```

Comment:
Instead of making `residues_selected` a list in the line below, you could do it directly here (residues_selected = list{atom.residue for atom in contact_atoms})